### PR TITLE
[PAY-946] Handle signed out users across DMs

### PR DESCRIPTION
**Pull request by piazzatron**
### Description
- Guard against accessing /messages pages
- Always show message button on profile page, but link to signup page if not logged in
- Remove block and unblock rows from StatsButton popover menu if not logged in (and also refactor this into a distinct component)
- Don't fetch permissions if not signed in!
Old:

https://github.com/AudiusProject/audius-client/assets/6780028/ae310020-9bc2-477d-b2d9-7ffb5381cbd5

New:

https://github.com/AudiusProject/audius-client/assets/6780028/bc51e766-3477-4395-90a6-ca7626edeac8

### Dragons

Need to confirm permissions do then get fetched after you log in...

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/web/src/pages/profile-page/ProfilePageProvider.tsx [hunk 1]
```diff
@@ -948,6 +960,7 @@ class ProfilePage extends PureComponent<ProfilePageProps, ProfilePageState> {
       onClickMobileOverflow: this.props.clickOverflow
     }
 
+    console.log({ chatPermissions: this.props.chatPermissions })
```
Comment (schottra) [inline packages/web/src/pages/profile-page/ProfilePageProvider.tsx:963]: Remove?

**Review by schottra — APPROVED** [review]
looks good, just remove the log

File: packages/web/src/pages/profile-page/ProfilePageProvider.tsx (hunk 1 above)
Comment (piazzatron) [reply packages/web/src/pages/profile-page/ProfilePageProvider.tsx:963]: thx

**Review by rickyrombo — APPROVED** [review]
Beautiful

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/pay-946

**Pull request merged**